### PR TITLE
[MRG] Add MatIO

### PR DESCRIPTION
**Pull request by massich**
- [x] Travis (linux)
  - [x] MatIO
  - [x] HDF5
- [x] Travis (mac)
  - [x] MatIO
  - [x] HDF5
- [x] appveyor
  - [x] MatIO
  - [x] HDF5

### CODE REVIEW

**Comment by agramfort** [comment]
you don't need all this code to test Matio

just put back MathsIO.C in OpenMEEGMaths_SOURCES

and that should it to test.

File: .travis.yml [hunk 1]
```diff
@@ -59,6 +27,7 @@ before_install:
       sudo apt-get update
       sudo apt-get install -y libc6:i386 libncurses5:i386 libstdc++6:i386
       sudo apt-get install -y lcov
+      sudo apt-get install -y libmatio libmatio-dev
```
Comment (agramfort) [inline .travis.yml]: you just need libmatio-dev

Comment (massich) [reply .travis.yml]: libmatio-dev only provides .a + includes.

Comment (massich) [reply .travis.yml]: .so belongs to libmatio2

File: .travis.yml [hunk 2]
```diff
@@ -59,6 +35,7 @@ before_install:
       sudo apt-get update
       sudo apt-get install -y libc6:i386 libncurses5:i386 libstdc++6:i386
       sudo apt-get install -y lcov
+      sudo apt-get install -y libmatio2 libmatio-dev
```
Comment (agramfort) [inline .travis.yml]: if you install both you will not know if you used the static or the dynamic lib

Comment (massich) [reply .travis.yml]: We are not linking statically against matio. Only dynamic. however, we do need the headers.

Comment (massich) [reply .travis.yml]: ldd should tell us. We really need to find a manner to test this

**Comment by agramfort** [comment]
but should link statically with matio if we can to avoid shipping the so

**Comment by massich** [comment]
I think we should be able to build independently of the shipping policy

**Comment by agramfort** [comment]
I agree but then we need to make that we actually test if we use libmatio.a or libmatio.so

**Comment by agramfort** [comment]
ok green !

**Comment by massich** [comment]
@agramfort I think we can merge it as it is, and then take care of [this error](https://ci.appveyor.com/project/massich/findmkl-openmeeg/build/1.0.221/job/aoajxsqjqdbp2479#L723). Some dll is doing something werid, but I cannot see it in the [dependency walk](https://ci.appveyor.com/api/buildjobs/aoajxsqjqdbp2479/artifacts/depends.txt)

**Comment by massich** [comment]
This PR got Merged with @agramfort +1

**Pull request merged**